### PR TITLE
BLOCKCHAIN-516 FE enable usage without any wallet

### PR DESCRIPTION
**Pull request by SzymonZiebaNeti**

### CODE REVIEW

File: src/redux/sagas/base.js [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
+import { blockchainActions, routeActions } from '../actions';
 
 function errorHandler(onFailure, worker) {
   function* errorHandledWorker(action) {
     try {
       yield* worker(action);
     } catch (e) {
       // eslint-disable-next-line no-console
-      console.error(e);
+      const isWallet = !e.stack.includes('Error: web3FromAddress: Unable to find injected');
```
Comment (jindra12) [inline src/redux/sagas/base.js:12]: This is very defensive and should be handled by reviewing the codebase and ensuring this doesn't happen to users. At the very least, this should be an "instanceof" check for error type if possible.

Comment (jindra12) [reply src/redux/sagas/base.js:12]: For instance, what if we upgrade polkadot library and  they change the error messages? This would stop working then.

Comment (jindra12) [reply src/redux/sagas/base.js:12]: If this is about accessing routes without an appropriate browser/wallet combination, I think react-router-dom library already has a way of validating a route, maybe try that?

File: src/components/CheckExtenstionWalletProvider/index.js [hunk 2]
```diff
@@ -59,15 +58,7 @@ export function CheckExtensionWalletProvider({ children }) {
...
-    return (
-      <GuidedSetupWrapper>
-        <NoWalletsDetectedInBrowser />
-      </GuidedSetupWrapper>
-    );
-  }
-
-  if (!walletAddress) {
+  if (!walletAddress && (userBlockchainAdressStorage || wallets[0]?.address)) {
     dispatch(blockchainActions.setUserWallet.success(userBlockchainAdressStorage || wallets[0].address));
```
Comment (jindra12) [inline src/components/CheckExtenstionWalletProvider/index.js:62]: Can you also fix this please? This should be an useEffect not a part of render

File: src/components/CheckExtenstionWalletProvider/index.js [hunk 3]
```diff
@@ -41,6 +40,16 @@ export function CheckExtensionWalletProvider({ children }) {
     || wallets === null
     || isUnsupportedBrowser === null;
 
+  useEffect(() => {
+    if (!walletAddress && (userBlockchainAdressStorage || (wallets?.length > 0 && wallets[0]?.address))) {
+      dispatch(
+        blockchainActions.setUserWallet.success(
+          userBlockchainAdressStorage || wallets[0].address,
```
Comment (DorianSternVukotic) [inline src/components/CheckExtenstionWalletProvider/index.js:47]: hmmm will have to see how this works with wallets that require you to connect again to the site

File: src/components/Home/ChangeRouter.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  const dispatch = useDispatch();
+
+  const history = useHistory();
+
+  useEffect(() => {
+    if (routeLink) {
+      setTimeout(() => {
+        history.push(routeLink);
+        dispatch(routeActions.changeRoute.call());
+      }, 2000);
```
Comment (DorianSternVukotic) [inline src/components/Home/ChangeRouter.js:18]: why is this on timeout, what is this doing?

File: src/redux/sagas/blockchain.js [hunk 5]
```diff
@@ -66,7 +66,7 @@ export function* subscribeWalletsSaga() {
       const wallets = await web3Accounts();
       emitter({ extensions, wallets });
     };
-    updateWallets();
+    setTimeout(updateWallets, 500);
```
Comment (DorianSternVukotic) [inline src/redux/sagas/blockchain.js:69]: why

**Comment by DorianSternVukotic** [comment]
Rebase

White screen of death on clicking wallet - send lld without wallet
When signing transaction without wallet i get notification no wallet detected, you need to login and then it automatically redirects. Show a message to user, thats fine, but dont auto redirect, add a login and cancel button when signing without wallet

File: src/components/ErrorModal/index.js [hunk 6]
```diff
@@ -21,14 +22,29 @@ function ErrorModal({ children }) {
...
+                  Ok
+                </Button>
+              )
+              : (
+                <div className={styles.buttons}>
+                  <Button
+                    secondary
+                    onClick={() => dispatch(blockchainActions.setErrorExistsAndUnacknowledgedByUser.call(false))}
+                  >
+                    Canel
```
Comment (DorianSternVukotic) [inline src/components/ErrorModal/index.js:41]: canCel

**Review by DorianSternVukotic — CHANGES_REQUESTED** [review]

**Pull request merged**